### ISSUE TITLE
Support for fish shell.

### ISSUE THREAD

**Issue by asmeurer** [conda/conda#540]
_Issue by **[latkins](https://github.com/latkins)** from Wednesday Jan 29, 2014 at 15:55 GMT_
_Originally opened as https://github.com/ContinuumIO/anaconda-issues/issues/75_

----

I was unable to follow the guide to set up anaconda with python3 (as found here: http://continuum.io/blog/anaconda-python-3) using fish shell. The problem comes from using activate. Would it be possible to add support for fish? Virtualenv manages this by including an activate.fish script.

**Comment by Nodd** [comment]
:+1: Fish is great.

**Comment by endrebak** [comment]
See https://pypi.python.org/pypi/conda-workon/0.1.0

Works great with fish.

**Comment by fabioramponi** [comment]
Here https://github.com/fabioramponi/fish-conda-virtualenv you can find two simple scripts that I wrote to activate/deactivate conda virtual environments from fish shell, by mimicking the behaviour of the bash/zsh conda activate/deactivate scripts.

These are written for personal use, so you could find some portability issue, but it works for me: I can write from a fish shell

```
>>> source (activate) <conda-env-name>
>>> source (deactivate)
```

**Comment by aldanor** [comment]
@fabioramponi Here's a working version: https://github.com/conda/conda/pull/545 (requires fish 2.2+ to properly work). I've been using it for over a month now with great success.

I really need to catch up and rebase that PR / add it to conda's manifest so it can get merged in...

**Comment by fabioramponi** [comment]
Great, I missed that thread: thanks @aldanor!

**Comment by aldanor** [comment]
Should probably be closed by #545

**Comment by msarahan** [comment]
Thanks.  Closing.
